parsers/IN.py

```diff
@@ -82,6 +119,44 @@ def fetch_production(
     return data
 
 
+def fetch_consumption(
+    zone_key: str,
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> dict:
+    """Fetches live consumption from government dashboard. Consumption is available per state and is then aggregated at regional level.
+    Data is not available for the following states: Ladakh (disputed territory), Daman & Diu, Dadra & Nagar Haveli, Lakshadweep"""
+    if target_datetime is not None:
+        raise NotImplementedError("This parser is not yet able to parse past dates")
+
+    total_consumption = 0
+    for state in STATES_MAPPING[zone_key]:
+        r = session.get(DEMAND_URL.format(state=state))
+        soup = BeautifulSoup(r.content, "html.parser")
+        try:
+            state_consumption = int(
+                soup.find(
+                    "span", attrs={"class": "value_DemandMET_en value_StateDetails_en"}
```

Comment:
Maybe we should have a specific exception in case we cannot find this attribute in the html. It would mean that we need to update our code to match the new attributes. Otherwise this would be silently ignored and will go under the radar.